### PR TITLE
Add more specifics about what Kubernetes does.

### PR DESCRIPTION
**Pull request by bgrant0607**
Answer questions people have about Kubernetes, such as:
http://stackoverflow.com/questions/28086732/what-does-kubernetes-actually-do

### CODE REVIEW

File: docs/whatisk8s.md [hunk 1]
```diff
@@ -79,11 +81,31 @@ Here are some key points:
 * **Resource utilization**:
     High efficiency and density.
 
+#### What can Kubernetes do?
+
+Kubernetes can schedule and run application containers on clusters of physical or virtual machines.
+
+It can also do much more than that.
+
+Kubernetes satisfies a number of common needs of applications running in production, such as [co-locating helper processes](user-guide/pods.md), [mounting storage systems](user-guide/volumes.md), distributing [secrets](user-guide/secrets.md), [application health checking](user-guide/production-pods.md#liveness-and-readiness-probes-aka-health-checks), [replicating application instances](user-guide/replication-controller.md), [horizontal auto-scaling](user-guide/horizontal-pod-autoscaler.md), [load balancing](user-guide/services.md), [rolling updates](user-guide/update-demo/), and [resource monitoring](user-guide/monitoring.md).
```
Comment (bgrant0607) [inline docs/whatisk8s.md]: Let me know if you think a bullet list would be better.

File: docs/whatisk8s.md [hunk 2]
```diff
@@ -79,11 +81,31 @@ Here are some key points:
...
+
+Even though Kubernetes provides a lot of functionality, there are always new scenarios that would benefit from new features. Ad hoc orchestration that is acceptable initially often requires robust automation at scale. Application-specific workflows can be streamlined to accelerate developer velocity. This is why Kubernetes was also designed to serve as a platform for building an ecosystem of components and tools to make it easier to deploy, scale, and manage applications.
+
+[Labels](user-guide/labels.md) empower users to organize their resources however they please. [Annotations](user-guide/annotations.md) enable users to decorate resources with custom information to facilitate their workflows and provide an easy way for management tools to checkpoint state.
+
+Additionally, the [Kubernetes control plane](admin/cluster-components.md) is built upon the same [APIs](api.md) that are available to developers and users. Users can write their own controllers, [schedulers](devel/scheduler.md), , etc., if they choose, with [their own APIs](design/extending-api.md) that can be targeted by a general-purpose [command-line tool](user-guide/kubectl-overview.md).
```
Comment (bgrant0607) [inline docs/whatisk8s.md]: AFAICT, we don't have a doc about the general "controller" concept.

File: docs/whatisk8s.md (hunk 1 above)
Comment (thockin) [inline docs/whatisk8s.md]: In this case I do

File: docs/whatisk8s.md (hunk 2 above)
Comment (thockin) [inline docs/whatisk8s.md]: spurios extra comma here: "scheduler.md), , etc.,"

**Comment by thockin** [comment]
Just a couple nits

**Comment by bgrant0607** [comment]
Thanks. Addressed feedback.

**Comment by kelseyhightower** [comment]
@bgrant0607 just read through the updates and I think we now have a complete picture of what Kubernetes in a single page. The links to more details is a nice touch. 

LGTM

**Comment by bgrant0607** [comment]
Thanks for taking a look, @kelseyhightower.

**Pull request merged**